### PR TITLE
[ADD][12.0] module mail_send_bcc to receive all mail sent by Odoo in bcc

### PR DESCRIPTION
**Pull request by legalsylvain**
Supersed https://github.com/OCA/server-tools/pull/1722. (bad repository)

Migrate a simple module from GRAP V8 incubator repository.

This module overload the send of email in odoo. For each mail sent, odoo will add, in the "Bcc" field, the email present in the "From" field.

Consequently, when the users are sending mail via Odoo to a partner (customer, supplier), they will receive a copy of the email.
Also, if the partner answer to the email, and your mail provider is handling correctly thread (as Gmail does), you'll have the whole history of the mails in your mailbox.

CC : @quentinDupont, @etobella, @pedrobaeza 

#GRAPOCA

### CODE REVIEW

File: mail_send_bcc/__manifest__.py [hunk 1]
```diff
@@ -0,0 +1,15 @@
...
+# @author: Sylvain LE GAL (https://twitter.com/legalsylvain)
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl.html).
+{
+    "name": "Mail - Send Email Bcc",
+    "summary": "Send to you in Bcc mode each mail sent by Odoo",
+    "version": "12.0.1.0.0",
+    "category": "Social Network",
+    "author": "GRAP"
+              "Odoo Community Association (OCA)",
+    "website": "https://www.github.com/OCA/server-tools",
```
Comment (etobella) [inline mail_send_bcc/__manifest__.py]: OCA/social

**Review by etobella — APPROVED** [review]
LGTM :+1: Just a small nitpicking (non blocking)

**Comment by JordiBForgeFlow** [comment]
@legalsylvain what do you think about this module supporting cc as well? Do you think that would be feasible?

**Comment by legalsylvain** [comment]
> @legalsylvain what do you think about this module supporting cc as well? Do you think that would be feasible?

Which module ? This one : https://github.com/OCA/social/tree/8.0/base_mail_bcc

If yes, that is different need. ``base_mail_bcc`` send allways to the same adresses. My module send to the sender. (from field)
``base_mail_bcc`` is useless in my situation. I have a DB with multi company enabled (50 companies) and adress should be dynamic.

So propose to keep both modules in the OCA. (even if in reality, the module base_mail_bcc is only available for V 8.)

**Comment by JordiBForgeFlow** [comment]
@legalsylvain I mean to add the ability to add people in CC and in BCC. Seems that this module only supports BCC. Right?

**Comment by legalsylvain** [comment]
@jbeficent : OK. I get it.
yes it is feasable, adding configuration in ``res.config.settings``. 
But it is not in the scope of this PR.  

I renamed the module to ``mail_send_copy`` and add a section in the Roadmap. Feel free to implement this feature.
Otherwise, is the PR OK ? 

regards.

**Review by quentinDupont — CHANGES_REQUESTED** [review]
Works fine for user :heavy_check_mark: 

Travis fail for a good reason : a `mail_tracking` test is failing with this module

[This is this test](https://github.com/OCA/social/blob/12.0/mail_tracking/tests/test_mail_tracking.py#L122).The test failed because we can send the email even if there is no partner email.
Indeed, with this module, we automatically add the BCC field, and so the email is sent even if there is not partner email.

Maybe we can check if there is a partner email before adding BCC ?

**Comment by legalsylvain** [comment]
> Travis fail for a good reason : a mail_tracking test is failing with this module

> This is this test.The test failed because we can send the email even if there is no partner email.
Indeed, with this module, we automatically add the BCC field, and so the email is sent even if there is not partner email.

> Maybe we can check if there is a partner email before adding BCC ?

Thanks a lot @quentinDupont  for your analysis. I'll take a look.

**Comment by legalsylvain** [comment]
Hi. 
There is an error for runbot, but the error is also present on 12.0 Branch. 
image

Ref : https://runbot.odoo-community.org/runbot/repo/https-github-com-oca-social-205

that's weird, because the runbot seems to have failed in the previous merged PR from @pedrobaeza available here : https://github.com/OCA/social/pull/602

but the bot seems to have ignored and merged.
I thought that using bot was preventing having main branches red.

Any idea ? @sbidoul maybe ? 

thanks !

**Comment by sbidoul** [comment]
> I thought that using bot was preventing having main branches red.

ocabot merge ignores runbot errors. Why is runbot red and travis green?

**Comment by legalsylvain** [comment]
Hi @sbidoul 

> ocabot merge ignores runbot errors. 

thanks for your answer. I didn't know.

> Why is runbot red and travis green?

No idea. there are some errors here : https://runbot3.odoo-community.org/runbot/static/build/3437583-462-fc2fc0/logs/job_10_test_base.txt but seems fully unrelated to my PR. 

```
AttributeError: module 'subprocess' has no attribute '_optim_args_from_interpreter_flags'

...

[91mnpm WARN saveError ENOENT: no such file or directory, open '/home/odoo/build/OCA/social/package.json'

...

Building wheel for pyldap (setup.py): finished with status 'error'
[91m  ERROR: Command errored out with exit status 1:
   command: /.repo_requirements/virtualenv/python3.5/bin/python -u -c 'import sys, setuptools, tokenize; sys.argv[0] = '"'"'/tmp/pip-install-kd21hkwp/pyldap/setup.py'"'"'; __file__='"'"'/tmp/pip-install-kd21hkwp/pyldap/setup.py'"'"';f=getattr(tokenize, '"'"'open'"'"', open)(__file__);code=f.read().replace('"'"'\r\n'"'"', '"'"'\n'"'"');f.close();exec(compile(code, __file__, '"'"'exec'"'"'))' bdist_wheel -d /tmp/pip-wheel-sgoo2or7
       cwd: /tmp/pip-install-kd21hkwp/pyldap/

...

  Building wheel for suds-jurko (setup.py): finished with status 'error'
[91m  ERROR: Command errored out with exit status 1:
   command: /.repo_requirements/virtualenv/python3.5/bin/python -u -c 'import sys, setuptools, tokenize; sys.argv[0] = '"'"'/tmp/pip-install-kd21hkwp/suds-jurko/setup.py'"'"'; __file__='"'"'/tmp/pip-install-kd21hkwp/suds-jurko/setup.py'"'"';f=getattr(tokenize, '"'"'open'"'"', open)(__file__);code=f.read().replace('"'"'\r\n'"'"', '"'"'\n'"'"');f.close();exec(compile(code, __file__, '"'"'exec'"'"'))' bdist_wheel -d /tmp/pip-wheel-7dqfofcs
       cwd: /tmp/pip-install-kd21hkwp/suds-jurko/

```

**Comment by sbidoul** [comment]
Looks like an issue with the latest setuptools. https://github.com/pypa/setuptools/issues/2357 seems related.

**Comment by sbidoul** [comment]
We can try this: https://github.com/OCA/maintainer-quality-tools/pull/675

**Comment by legalsylvain** [comment]
now green. Thanks @sbidoul !

**Review by quentinDupont — APPROVED** [review]

**Comment by legalsylvain** [comment]
Hi @OCA/social-maintainers : this PR could be merged IMO.

**Comment by sebastienbeau** [comment]
/ocabot merge nobump 
Thanks for your work @legalsylvain

**Comment by rafaelbn** [comment]
@JordiBForgeFlow @legalsylvain FYI: https://github.com/OCA/social/tree/15.0/mail_composer_cc_bcc

**Comment by legalsylvain** [comment]
Thanks ! I didn't know ! The module looks way more  complex than the current one. (with full overwrite of [odoo functions](https://github.com/OCA/social/blob/15.0/mail_composer_cc_bcc/models/mail_mail.py#L28))

but looks  promising and is more complete !

**Pull request merged**